### PR TITLE
Pcm 2302

### PR DESCRIPTION
**Pull request by maxwellmooney13**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -139,4 +139,49 @@ export const logPendingSession = function (
 
 export function getBareJid (sdk: GenesysCloudWebrtcSdk) {
   return sdk._streamingConnection.config.jid;
+}
+
+export class HeadsetChangesQueue {
```
Comment (jensengar) [inline src/utils.ts:144]: This probably belongs in its own file in the headsets folder

File: src/sessions/softphone-session-handler.ts [hunk 2]
```diff
@@ -171,11 +171,11 @@ export class SoftphoneSessionHandler extends BaseSessionHandler {
     /* It really should only matter for connected calls as ended calls should have the logic already in place in SVH */
     if (this.isConnectedState(callState) && session) {
       if (callState.muted !== previousCallState?.muted) {
-        this.sdk.headset.setMute(callState.muted);
+        await HeadsetChangesQueue.queueHeadsetChanges(() => this.sdk.headset.setMute(callState.muted));
```
Comment (jensengar) [inline src/sessions/softphone-session-handler.ts:174]: So I don't think you need to await the headsetChangesQueue. See Example: https://stackblitz.com/edit/typescript-1yxa9i?file=index.ts

Was this due to linter complaints of dangling promises?

**Review by jensengar — CHANGES_REQUESTED** [review]

File: src/sessions/softphone-session-handler.ts (hunk 2 above)
Comment (maxwellmooney13) [reply src/sessions/softphone-session-handler.ts:174]: I think this was more so just me taking precautionary measures to ensure the queue is in the order we expect.  But looking at your example, it does look unnecessary.  I'll remove it and test to see if it's still fine

**Review by jensengar — APPROVED** [review]

**Review by hjon — APPROVED** [review]

**Comment by hjon** [comment]
Nit: update changelog. 😄

File: src/sessions/softphone-session-handler.ts (hunk 2 above)
Comment (maxwellmooney13) [reply src/sessions/softphone-session-handler.ts]: Okay yeah! It doesn't look like the await is needed

**Pull request merged**